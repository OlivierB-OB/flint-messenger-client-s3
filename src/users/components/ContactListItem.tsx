import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import React, { Fragment } from 'react';
import { IUserInfo } from '../types';
import { UserAvatar } from './UserAvatar';
import { UserName } from './UserName';

export interface IContactListItemProps {
  info: IUserInfo;
}

export function ContactListItem({ info }: IContactListItemProps) {
  const { status } = info;
  return (
    <Fragment>
      <ListItemAvatar>
        <UserAvatar info={info} />
      </ListItemAvatar>
      <ListItemText primary={<UserName info={info} />} secondary={status} />
    </Fragment>
  );
}
