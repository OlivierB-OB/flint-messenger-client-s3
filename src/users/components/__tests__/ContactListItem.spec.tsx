import React from 'react';
import { shallow } from 'enzyme';
import { ContactListItem } from '../ContactListItem';

describe('ContactListItem', () => {
  it('should show the contact with avatar name and status', async () => {
    const info: any = {
      firstName: 'foo',
      lastName: 'bar',
      status: 'available',
    };
    const component = shallow(<ContactListItem info={info} />);
    expect(component).toMatchSnapshot();
  });
});
