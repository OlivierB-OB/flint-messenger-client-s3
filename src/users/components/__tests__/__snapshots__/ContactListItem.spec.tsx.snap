// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactListItem should show the contact with avatar name and status 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <UserAvatar
      info={
        Object {
          "firstName": "foo",
          "lastName": "bar",
          "status": "available",
        }
      }
    />
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    primary={
      <UserName
        info={
          Object {
            "firstName": "foo",
            "lastName": "bar",
            "status": "available",
          }
        }
      />
    }
    secondary="available"
  />
</Fragment>
`;
